drivers/storage/portworx/portworx_test.go

```diff
@@ -471,11 +473,26 @@ func TestSetDefaultsOnStorageClusterOnEKS(t *testing.T) {
 	require.Equal(t, "true", check)
 
 	// Check cloud storage spec
-	expectedCloudStorage := "type=gp3,size=150"
+	require.Nil(t, cluster.Spec.Storage)
 	require.NotNil(t, cluster.Spec.CloudStorage.DeviceSpecs)
-	require.Contains(t, *cluster.Spec.CloudStorage.DeviceSpecs, expectedCloudStorage)
+	expectedCloudStorageDevice := "type=gp3,size=150"
+	require.Contains(t, *cluster.Spec.CloudStorage.DeviceSpecs, expectedCloudStorageDevice)
 	require.True(t, cluster.Spec.Kvdb.Internal)
 
+	// Check cloud storage spec when capacity spec specified
+	cluster.Spec = corev1.StorageClusterSpec{
+		CloudStorage: &corev1.CloudStorageSpec{
+			CapacitySpecs: []corev1.CloudStorageCapacitySpec{{
+				MinCapacityInGiB: 500,
+			}},
+		},
+	}
+	err = driver.SetDefaultsOnStorageCluster(cluster)
```

Comment:
can you get the px pod spec and validate if capacity spec is really used.

Reply:
I found we have a full unit test coverage for that here https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/deployment_test.go#L770

Reply:
sorry updated, this one https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/deployment_test.go#L1033

Reply:
the code path is slightly different, as you set deviceSpec in your code. I think even if you don't add this check it will still work, what happens was if capacitySpecs is not empty deviceSpecs were still set, however capacitySpecs take higher priority than deviceSpec when generating storage pod spec. hence it should work. correct?

if len(toUpdate.Spec.CloudStorage.CapacitySpecs) == 0 &&
			toUpdate.Spec.CloudStorage.DeviceSpecs == nil